torch/nn/_intrinsic/qat/modules/conv_fused.py

```diff
@@ -0,0 +1,265 @@
+from torch.nn._intrinsic import ConvBn2d as NNConvBn2d
+from torch.nn._intrinsic import ConvBnReLU2d as NNConvBnReLU2d
+from torch.nn._intrinsic import ConvReLU2d as NNConvReLU2d
+from ....qat.modules.conv import Conv2d as QATConv2d
+from torch.nn import Parameter
+import torch.nn.functional as F
+
+
+class ConvBn2d(Conv2d):
+    r"""
+    A ConvBn2d module is a module fused from Conv2d and BatchNorm2d,
+    attached with FakeQuantize modules for both output activation and weight,
+    used in quantization aware training.
+
+    We combined the interface of :class:`torch.nn.Conv2d` and
+    :class:`torch.nn.BatchNorm2d`.
+
+    Implementation details: https://arxiv.org/pdf/1806.08342.pdf section 3.2.2
+
+    Similar to :class:`torch.nn.Conv2d`, with FakeQuantize modules initialized
+    to default.
+
+    Attributes:
+        freeze_bn:
+        observer: fake quant module for output activation, it's called observer
+            to align with post training flow
+        weight_fake_quant: fake quant module for weight
+
+    """
+    __FLOAT_MODULE = NNConvBn2d
+
+    def __init__(self,
+                 # conv2d args
+                 in_channels, out_channels, kernel_size, stride=1,
+                 padding=0, dilation=1, groups=1,
+                 bias=True, padding_mode='zeros',
+                 # bn args
+                 # num_features: enforce this matches out_channels before fusion
+                 eps=1e-05, momentum=0.1,
+                 # affine: enforce this is True before fusion?
+                 # tracking_running_stats: enforce this is True before fusion
+                 # args for this module
+                 freeze_bn=False,
+                 activation_fake_quant=None,
+                 weight_fake_quant=None):
+        super(ConvBn2d, self).__init__(in_channels, out_channels, kernel_size,
+                                       stride, padding, dilation, groups, bias, padding_mode)
+        self.eps = eps
+        self.momentum = momentum
+        self.freeze_bn = freeze_bn if self.training else True
+        self.num_features = out_channels
+        self.gamma = Parameter(torch.Tensor(out_channels))
+        self.beta = Parameter(torch.Tensor(out_channels))
+        self.affine = True
+        self.track_running_stats = True
+        self.register_buffer('running_mean', torch.zeros(out_channels))
+        self.register_buffer('running_var', torch.ones(out_channels))
+        self.register_buffer('num_batches_tracked', torch.tensor(0, dtype=torch.long))
+        self.observer = activation_fake_quant()
+        self.weight_fake_quant = weight_fake_quant()
+        self.reset_bn_parameters()
+
+    def reset_running_stats(self):
+        self.running_mean.zero_()
+        self.running_var.fill_(1)
+        self.num_batches_tracked.zero_()
+
+    def reset_bn_parameters(self):
+        self.reset_running_stats()
+        init.uniform_(self.gamma)
+        init.zeros_(self.beta)
+
+    def reset_parameters(self):
```

Comment:
You could do:
```python
def reset_parameters(self):
        super(ConvBn2d, self).reset_parameters()
        # A hack to avoid resetting on undefined parameters
        if hasattr(self, 'gamma'):
            self.reset_bn_parameters()
```